addons/account/models/account_payment_method.py

```diff
@@ -103,7 +103,7 @@ class AccountPaymentMethodLine(models.Model):
                             "'|', ('user_type_id', '=', %s), ('id', '=', parent.default_account_id)]"
                             % self.env.ref('account.data_account_type_current_assets').id
     )
-    journal_id = fields.Many2one(comodel_name='account.journal', ondelete="set null")
+    journal_id = fields.Many2one(comodel_name='account.journal', ondelete="cascade")
```

Comment:
Why is `journal_id` not required again?

Reply:
Because we had to be able to delete old lines, but couldn't lose the information about them on the payments (like account/...)
So we had to be able to have lines existing without journals in this specific case.